pkg/hub/localizer/localizer.go

```diff
@@ -122,7 +133,282 @@ func (l *Localizer) handleGlobalization(glob *appsapi.Globalization) error {
+	var glos []*appsapi.Globalization
+
+	globalizations, err := l.globLister.List(labels.NewSelector())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, glo := range globalizations {
+		if IsFeedMatchObject(glo.Spec.Feed, unstructured) {
+			glos = append(glos, glo)
+		}
+	}
+
+	return glos, nil
+}
+
+func (l *Localizer) getLocalizationByResource(namespace string, unstructured *unstructured.Unstructured) ([]*appsapi.Localization, error) {
+	var locs []*appsapi.Localization
+
+	localizations, err := l.locLister.Localizations(namespace).List(labels.NewSelector())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, loc := range localizations {
+		if IsFeedMatchObject(loc.Spec.Feed, unstructured) {
+			locs = append(locs, loc)
+		}
+	}
+
+	return locs, nil
+}
+
+func (l *Localizer) getGlobalizationByChartRef(chartRef appsapi.ChartReference) ([]*appsapi.Globalization, error) {
+	var glos []*appsapi.Globalization
+
+	globalizations, err := l.globLister.List(labels.NewSelector())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, glo := range globalizations {
+		if IsFeedMatchHelmChart(glo.Spec.Feed, chartRef) {
+			glos = append(glos, glo)
+		}
+	}
+
+	return glos, nil
+}
+
+func (l *Localizer) getLocalizationByChartRef(namespace string, chartRef appsapi.ChartReference) ([]*appsapi.Localization, error) {
+	var locs []*appsapi.Localization
+
+	localizations, err := l.locLister.Localizations(namespace).List(labels.NewSelector())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, loc := range localizations {
+		if IsFeedMatchHelmChart(loc.Spec.Feed, chartRef) {
+			locs = append(locs, loc)
+		}
+	}
+
+	return locs, nil
+}
+
+func (l *Localizer) executionOverride(overriderType appsapi.OverrideType, overrides []appsapi.OverrideRule, raw []byte) ([]byte, error) {
+	output := raw
+	var err error
+	for _, override := range overrides {
+		if overriderType == appsapi.GenericOverride {
+			patcher, err := newPatcher(override, output)
+			if err != nil {
+				return nil, err
+			}
+			output, err = patcher.Apply()
+			if err != nil {
+				return nil, err
+			}
+		} else if overriderType == appsapi.HelmOverride {
+			output, err = MergeChartValues([]byte(override.OverrideValue["values"]), output)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return output, nil
+}
+
+func IsFeedMatchObject(feed appsapi.Feed, unstructured *unstructured.Unstructured) bool {
+	if len(feed.Name) > 0 {
+		return feed.APIVersion == unstructured.GetAPIVersion() &&
+			feed.Kind == unstructured.GetKind() &&
+			feed.Name == unstructured.GetName() &&
+			feed.Namespace == unstructured.GetNamespace()
+	}
+
+	if feed.FeedSelector != nil {
+		feedSelector, err := metav1.LabelSelectorAsSelector(feed.FeedSelector)
+		if err != nil {
+			return false
+		}
+
+		return feedSelector.Matches(labels.Set(unstructured.GetLabels()))
+	}
+
+	return false
+}
+
+func IsFeedMatchHelmChart(feed appsapi.Feed, chartReference appsapi.ChartReference) bool {
```

Comment:
what if we specify `feed.Selector` here?

Reply:
For every `Localization` and `Globalization`, you could find a label, like `some-uid-string=Namespace`.

Reply:
maybe it's the old version problem, using the label selector solution now